python/ray/serve/http_proxy.py

```diff
@@ -110,18 +112,19 @@ async def __call__(self, scope, receive, send):
         http_body_bytes = await self.receive_http_body(scope, receive, send)
 
         headers = {k.decode(): v.decode() for k, v in scope["headers"]}
-        request_metadata = RequestMetadata(
-            get_random_letters(10),  # Used for debugging.
+
+        handle = RayServeHandle(
+            self.controller_handle,
             endpoint_name,
-            TaskContext.Web,
-            http_method=scope["method"].upper(),
-            call_method=headers.get("X-SERVE-CALL-METHOD".lower(), "__call__"),
+            sync=True,
```

Comment:
we also want this to be async because we're in an asyncio loop here, but that will also be handled if we use `get_handle`